pkg/block/metrics.go

```diff
@@ -23,12 +23,20 @@ func (m *MetricsAdapter) InnerAdapter() Adapter {
 }
 
 func (m *MetricsAdapter) Put(ctx context.Context, obj ObjectPointer, sizeBytes int64, reader io.Reader, opts PutOpts) (*PutResponse, error) {
-	ctx = httputil.SetClientTrace(ctx, m.adapter.BlockstoreType())
+	blockstoreType, err := m.adapter.BlockstoreType(obj.StorageID)
```

Comment:
Just a thought:
If BlockAdapter will wrap MetricsAdapter instead of the other way around, you'd probably won't need to provide the storageID to BlockstoreType

Reply:
What does it mean more "technical"?
Switching the wrapping order might release us from the need to pass the storageID to the blockstore type.
This PR handles the issue of "Add StorageID to adapter metadata functions". What is the relevant issue for this comment if not this one?

Reply:
Look at the example of "statsCollector"